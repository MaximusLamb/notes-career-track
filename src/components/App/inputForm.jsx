import React from 'react';
import propTypes from 'prop-types';

const inputForm = ({ username, onChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <input type='text' value={username} onChange={onChange}/>
    <button>Submit</button>
  </form>
);

inputForm.propTypes = {
  username: propTypes.string.isRequired,
  onChange: propTypes.func.isRequired,
  onSubmit: propTypes.func.isRequired
};

export default inputForm;
