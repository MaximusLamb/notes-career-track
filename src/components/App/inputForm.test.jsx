import React from 'react';
import { shallow } from 'enzyme';
import inputForm from './inputForm';

describe('inputForm component', () => {
  it('renders inputForm', () => {
    const wrapper = shallow(<inputForm username="someBroski" onChange={() => {}} onSubmit={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
