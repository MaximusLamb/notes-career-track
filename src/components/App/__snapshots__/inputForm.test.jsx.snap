// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputForm component renders inputForm 1`] = `
<inputForm
  onChange={[Function]}
  onSubmit={[Function]}
  username="someBroski"
/>
`;
